### PR TITLE
Lockfile path tests (follow-up)

### PR DESCRIPTION
**Pull request by Ifropc**
Addressing comments by [Weihang](https://github.com/rust-lang/cargo/pull/14326#issuecomment-2293664459) and [Ed](https://github.com/rust-lang/cargo/pull/14326#issuecomment-2293671992) from #14326

### CODE REVIEW

File: tests/testsuite/lockfile_path.rs
Comment (weihanglo) [inline tests/testsuite/lockfile_path.rs:1]: While this patch looks like, we'd better follow [the atomic commit patterns](https://doc.crates.io/contrib/process/working-on-cargo.html#submitting-a-pull-request). That is, nightly gate tests are in one commit. Removing `make_exec` helper in the other.

Comment (weihanglo) [reply tests/testsuite/lockfile_path.rs:1]: Could also be two separate PR, as they are not really related. I am fine with one PR though.

Comment (Ifropc) [reply tests/testsuite/lockfile_path.rs:1]: Still learning the best way of doing commits for rust ecosystem 👍🏼 
I will just split it into 2 separate commits, I think logically one PR should be just fine

**Review by weihanglo — APPROVED** [review]
Looks better. Thank you!

@bors r+

**Comment by weihanglo** [comment]
I forgot. We also need an unstable doc. Something like <https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#precise-pre-release>. See the new tracking issue <https://github.com/rust-lang/cargo/issues/14421>

**Pull request merged**